### PR TITLE
 dev/core#5029 somewhat fix save template on sms trait

### PR DESCRIPTION
**Pull request by eileenmcnaughton**
Overview
----------------------------------------
This at least ensures it is possible to update the template & also fixes the notice & at least one js error

It still seems a little funky but I think it is closer to right & the gap is not on not doing enough hiding rather than not showing what is needed.

I also fixed the save to use the api & pulled in https://github.com/civicrm/civicrm-core/pull/29505 and https://github.com/civicrm/civicrm-core/pull/29578 because I allowed the permissions to be FALSE on save so wanted this protection.

Note I am targetting the just-cut 5.72 rc as the SMS form has been substantially re-written in 5.72 so my goal was to resolve the other known issues in 5.72 rather than spread changes to this form over more than one release

Before
----------------------------------------
- the update template box never shows 
- smarty notice
- js error when clearing the selected template (because the subject field does not exist)

After
----------------------------------------
Above fixed - still doesn't seem to hide the update box when it should - but that feels like the lesser evil

Technical Details
----------------------------------------
@demeritcowboy this is the issue you reported + a bit more. Note that the form has been re-written since you last looked - the "to" box now uses apiv4 ajax so keep you eye out for anything else odd

Comments
----------------------------------------

### CODE REVIEW

File: templates/CRM/Mailing/Form/InsertTokens.tpl [hunk 1]
```diff
@@ -102,31 +109,34 @@ function showSaveUpdateChkBox(prefix) {
...
+      document.getElementById(prefix + "updateDetails").style.display = "none";
+    }
+    else {
+      document.getElementById(prefix + "updateDetails").style.display = "block";
+    }
   }
 }
 
 function selectValue( val, prefix) {
+  console.log('here');
```
Comment (demeritcowboy) [inline templates/CRM/Mailing/Form/InsertTokens.tpl:122]: here

**Comment by demeritcowboy** [comment]
It seems ok just when doing Actions -> "add outbound sms" from a view contact page the "update template" box is visible until you start typing.

File: templates/CRM/Mailing/Form/InsertTokens.tpl (hunk 1 above)
Comment (eileenmcnaughton) [reply templates/CRM/Mailing/Form/InsertTokens.tpl:122]: `console.log('lol')`

**Comment by demeritcowboy** [comment]
Also not from this PR since I think I've seen it before but the Continue link on mass sms when you save a draft always creates a duplicate sms.

**Comment by eileenmcnaughton** [comment]
@demeritcowboy I'm not seeing that disappear - not even when it probably should - that old js code broke my brain a bit tho

image

**Comment by demeritcowboy** [comment]
It won't disappear if you've actually selected a template. That seems fine. It's that it shouldn't be there to start out with. But not a blocker since it didn't work at all before.

**Comment by eileenmcnaughton** [comment]
@demeritcowboy yeah - I'm so confused by that code - let's accept better

**Pull request merged**